rom/dev/src/flow/warm_reset.rs

```diff
@@ -31,12 +31,23 @@ impl WarmResetFlow {
     pub fn run(env: &mut RomEnv) -> CaliptraResult<()> {
         cprintln!("[warm-reset] ++");
 
-        // Check if previous Cold-Reset was successful.
-        if cfi_launder(env.data_vault.rom_cold_boot_status()) != ColdResetComplete.into() {
-            cprintln!("[warm-reset] Previous Cold-Reset was not successful.");
-            return Err(CaliptraError::ROM_WARM_RESET_UNSUCCESSFUL_PREVIOUS_COLD_RESET);
+        // Check if previous [Cold | Update] reset was successful.
+        if cfi_launder(env.data_vault.rom_cold_boot_status()) != ColdResetComplete.into()
+            && cfi_launder(env.data_vault.rom_update_reset_status()) != UpdateResetComplete.into()
```

Comment:
I don't think this logic is right. If rom_cold_boot_status() == ColdResetComplete, then we'll never go into this block...

Reply:
These values can have three states: Default, Started, and Complete.

I think we want `rom_cold_boot_status() != ColdResetComplete || rom_update_reset_status() == UpdateResetStarted`

Reply:
Or maybe the slightly less glitch prone:

```
rom_cold_boot_status() != ColdResetComplete || (rom_update_reset_status() != UpdateResetDefaultValue && rom_update_reset_status() != UpdateResetCompleted)
```

Reply:
Fixed.